### ISSUE TITLE
Convert markdown documentation to rst

### ISSUE THREAD

**Issue by cgeorgii** [haskell/haskell-language-server#3663]
## Motivation

Markdown is great but lacks some interesting features:
- Importing source code with `literalInclude`
- Admonition boxes
- etc.

## Proposed solution
Ideally all markdown files in the `docs/` directory could be converted to reStructuredText.

## Additional context
This issue originated from a [comment](https://github.com/haskell/haskell-language-server/pull/3655#discussion_r1229835773).

**Comment by michaelpj** [comment]
I don't think we should do this wholesale: the markdown files are easier for people to contribute to. But using RST where it provides useful additional power is sensible.
